### PR TITLE
Add basic github action to migrate from travis & CodeQL Analysis

### LINKED ISSUE #74: Switching CI to Github Actions
## Expected Behavior
Travis is beautiful and nice and does not queue the jobs for almost hour.

## Actual Behavior
Travis queues the jobs for almost an hour if the job is triggered during a kind of "peak time".

## How to fix
It is possible to get an idea of the current number of jobs running on travis via https://www.traviscistatus.com/
However, it seems like some migrations is ongoing from travis-ci.org (which will be closed on december, 31st) to travis-ci.com and that moving to travis-ci.com will not help that much. Following [this discussion](https://travis-ci.community/t/org-com-migration-unexpectedly-comes-with-a-plan-change-for-oss-what-exactly-is-the-new-deal/10567/2), I think it is better to just move [to Github Actions](https://github.com/features/actions).

### PR DESCRIPTION
**Pull request by ryarnyah**
Fixes #74

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ feat/use-github-actions ]
```
Comment (StephenSorriaux) [inline .github/workflows/codeql-analysis.yml:16]: Would this need a change before merging this PR?

**Review by StephenSorriaux — APPROVED** [review]
Awesome. This is pure gold. Thank you a lot.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (ryarnyah) [reply .github/workflows/codeql-analysis.yml:16]: Done

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
```
Comment (StephenSorriaux) [inline .github/workflows/codeql-analysis.yml:20]: Just wondering: is this `schedule` required by CodeQL (I did not read their documentation yet :grimacing: )?

Comment (ryarnyah) [reply .github/workflows/codeql-analysis.yml:20]: Yep if you want your code scanned every day for new vulns (it's random of course :D )

Comment (StephenSorriaux) [reply .github/workflows/codeql-analysis.yml:20]: Perfect, thanks again!

**Pull request merged**